### PR TITLE
User leaves a discussion

### PR DESCRIPTION
**Pull request by gdaguin**
# Description

User can now leave a discussion

Fixes # (issue)

## Type of change

- New feature (non-breaking change which adds functionality)

## Material used:

Please delete options that are not relevant.
- iOS (Smartphone/Tablet) : iPhone 15 Pro Max

## How Has This Been Tested?

Go into a discussion (that you have created or not), and go into the details of it. Then leave the discussion.

### Patrol

Please list here the tests created in Patrol for this pull request.

## Checklist:

- My code follows the style guidelines of this project
- I have performed a self-review of my own code
- I have commented my code, particularly in hard-to-understand areas
- I have made corresponding changes to the documentation
- My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature works
- New and existing unit tests pass locally with my changes
- Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: lib/ui/views/messenger/layouts/messenger_discussion_page.dart [hunk 1]
```diff
@@ -93,9 +95,13 @@ class MessengerDiscussionPage extends ConsumerWidget {
                   discussionAddress: discussionAddress,
                 ),
               ),
-              _MessageSendForm(
-                discussionAddress: discussionAddress,
-              ),
+              if (discussion.value != null &&
+                  discussion.value!.membersPubKeys.any(
+                    (element) => element == selectedContact?.publicKey,
+                  )) // User can only send a message when he is still is the discussion
```
Comment (redDwarf03) [inline lib/ui/views/messenger/layouts/messenger_discussion_page.dart]: typo "in"

File: lib/ui/views/messenger/bloc/discussion_details_form.dart [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+              membersPubKeys: state.members
+                  .where((element) => element != selectedContact.publicKey)
+                  .toList(),
+              discussionName: state.name,
+              adminAddress: selectedAccount.lastAddress!,
+              serviceName: selectedAccount.name,
+              session: session,
+              adminKeyPair: keyPair,
+              owner: selectedAccount,
+              updateSCAESKey:
```
Comment (redDwarf03) [inline lib/ui/views/messenger/bloc/discussion_details_form.dart:118]: It's not linked to this PR, but have you implement the update of AESKey when an admin is removed from admins list ?

File: lib/ui/views/messenger/bloc/discussion_details_form.dart [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+  void init(Discussion discussion) {
+    setName(discussion.name ?? '');
+    setAddress(discussion.address);
+    _initMembers(discussion.membersPubKeys);
+    _initAdmins(discussion.adminsPubKeys);
+  }
+
+  void _initMembers(List<String> members) {
+    for (final member in members) {
```
Comment (redDwarf03) [inline lib/ui/views/messenger/bloc/discussion_details_form.dart:38]: void _initMembers(List<String> _members) {
    members.addAll(_members);
}

File: lib/ui/views/messenger/bloc/discussion_details_form.dart [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+      addMember(member);
+    }
+  }
+
+  void _initAdmins(List<String> admins) {
+    for (final admin in admins) {
```
Comment (redDwarf03) [inline lib/ui/views/messenger/bloc/discussion_details_form.dart:44]: void _initMembers(List<String> _admins) {
    admins.addAll(_admins);
}

File: lib/ui/views/messenger/bloc/discussion_details_form.dart (hunk 2 above)
Comment (gdaguin) [reply lib/ui/views/messenger/bloc/discussion_details_form.dart:118]: Nope, but I'm going to make it

File: lib/ui/views/messenger/bloc/discussion_details_form.dart (hunk 3 above)
Comment (gdaguin) [reply lib/ui/views/messenger/bloc/discussion_details_form.dart:38]: It's all about calling the addMember or addAdmin sub-functions, that are updating the state of the notifier. So I think that I am going to not change anything.

File: lib/ui/views/messenger/bloc/discussion_details_form.dart (hunk 4 above)
Comment (gdaguin) [reply lib/ui/views/messenger/bloc/discussion_details_form.dart:44]: It's all about calling the addMember or addAdmin sub-functions, that are updating the state of the notifier. So I think that I am going to not change anything.

**Comment by gdaguin** [comment]
@redDwarf03 all good to me.

File: lib/infrastructure/repositories/messenger_repository.dart [hunk 5]
```diff
@@ -155,51 +155,38 @@ class MessengerRepository
...
-              ),
-              'fr': const PushNotification(
-                title: 'Archethic',
-                body: "Vous avez été supprimé d'une discussion",
-              ),
-            },
-            transactionType:
-                MessengerConstants.notificationTypeNewDiscussionDeletion,
-          );
+          // https://stackoverflow.com/questions/21826342/how-do-i-combine-two-lists-in-dart with If you want to merge two lists and remove duplicates
```
Comment (redDwarf03) [inline lib/infrastructure/repositories/messenger_repository.dart:162]: add all your elements with doublon in a list and do that to remove doublons:
 xxxxx.toSet().toList(),

**Review by redDwarf03 — APPROVED** [review]

**Pull request merged**